fraud/registry.go

```diff
@@ -0,0 +1,29 @@
+package fraud
+
+import (
+	"fmt"
+	"sync"
+)
+
+var unmarshalersLk = sync.RWMutex{}
+var defaultUnmarshalers = map[ProofType]ProofUnmarshaler{}
+
+func init() {
+	Register(BadEncoding, UnmarshalBEFP)
+}
+
+func Register(proofType ProofType, unmarshaler ProofUnmarshaler) {
+	unmarshalersLk.Lock()
+	unmarshalersLk.Unlock()
+	defaultUnmarshalers[proofType] = unmarshaler
+}
+
+func GetUnmarshaler(proofType ProofType) (ProofUnmarshaler, error) {
```

Comment:
What about `Unmarshal` func that takes both type and method> There is no real usage for an `unmarhaller` besides calling it, so we can call it right away.

Reply:
May be let's do `func Unmarshal(p proofType, data []byte) (Proof, error)` then? 

> func that takes both type and method

We should somehow get this method. I don't like calling a global mutex inside `fraud.Service` methods.

Reply:
LgTM